### ISSUE TITLE
Ingame viewer counter

### ISSUE THREAD

**Issue by AssassinTee** [teeworlds/teeworlds#2283]
I would really like an ingame-viewer-counter. I already made a test branch with it. But before I PR anything i just wanted to show it and want some feedback of the idea in general. You could easily add a variable to turn it on and off.

Ignore the uglyness of the icon, it's just a placeholder.

![example](https://i.imgur.com/iYtpX4c.png)

**Comment by Dune-jr** [comment]
I think this looks good.

**Comment by AssassinTee** [comment]
After a bit of struggeling, i _could_ add a twitch viewercount, too.
Teeworlds is just missing a http client

**Comment by Dune-jr** [comment]
That kind of feature would add dependencies and definitely be a bloat in Teeworlds imo. Best left to custom clients indeed :slightly_smiling_face:

**Comment by sepulzera** [comment]
A spectated-counter is no ingame related information and shouldn't be displayed there to avoid distraction.

**Comment by AssassinTee** [comment]
@sepulzera i can argue with that, the ingame viewer counter may have ingame related information in vanilla and mods:
- in zCatch you can see the catched people 
- in clanwars and tournaments you see the how many people are watching (I would really like to know this stuff)
- When showing something (for example in training), you can ensure that everyone is watching

I know some of that arguments are weak, but you could easily hide the spec-count behind a config variable and therefore avoid the distraction.

**Comment by Sonix-** [comment]
I agree with @sepulzera, perhaps it could be placed in a different spot as it is not a core game information.. maybe scoreboard or scoreboard view?

**Comment by AssassinTee** [comment]
My newest Idea concerning this is much simpler and may be a better spot:
On the bottom of the scoreboard is already a spectator count:
for example:

```
Spectators(1): nameless tee
```
so if anyone is spectating you, you could just make it

`Spectators(1/1): nameless tee` (`Spectators(<watchers>/<spectators>)`)

**Comment by jxsl13** [comment]
Show counter when the scoreboard is shown. Otherwise irrelevant in game information, if not in the scoreboard UI.

**Comment by Robyt3** [comment]
I like the idea of integrating this into the scoreboard, but the Watchers/Spectators format is too confusing IMO. I'd prefer something more explicit like:
```
Spectators(3): a, b, c
Watching you(2): d, e
```
With watchers only shown in their list and not in the spectators list. And only showing "Watching you" if more than 0.

**Comment by Sonix-** [comment]
Instead of creating another Line for the ones watching you, just show them first in the spectators list, before the other spectators. Generally you dont care how many are watching you.. but who

**Comment by AssassinTee** [comment]
Spectators specking you are already marked in yellow so is this usefull?

**Comment by Dune-jr** [comment]
Clearly the controversy here is whether this information should be shown at all times
